src/middleware/navigation/has.trust.middleware.ts

```diff
@@ -28,6 +29,49 @@ const hasTrustWithId = (req: Request, res: Response, next: NextFunction, url: st
   }
 };
 
+const hasTrusteeWithId = (req: Request, res: Response, next: NextFunction, url: string): void => {
+  try {
+    const trustId = req.params[ROUTE_PARAM_TRUST_ID];
+    const trusteeId = req.params[ROUTE_PARAM_TRUSTEE_ID];
+    const trusteeType = req.params[ROUTE_PARAM_TRUSTEE_TYPE];
+    const appData: ApplicationData = getApplicationData(req.session);
+
+    const isTrustPresent = appData[TrustKey]?.some(
+      (trust) => trust.trust_id === trustId,
+    );
+
+    if (!isTrustPresent) {
+      logger.infoRequest(req, NavigationErrorMessage);
+
+      return res.redirect(url);
+    }
+
+    let hasTrusteeForChange = false;
+    if (trusteeId && trusteeType) {
+      switch (trusteeType as TrusteeType){
+          case TrusteeType.HISTORICAL:
+            hasTrusteeForChange = (appData[TrustKey]?.find(t => t.trust_id === trustId)?.HISTORICAL_BO ?? []).some(t => t.id === trusteeId);
+            break;
+          case TrusteeType.INDIVIDUAL:
+            hasTrusteeForChange = (appData[TrustKey]?.find(t => t.trust_id === trustId)?.INDIVIDUALS ?? []).some(t => t.id === trusteeId);
+            break;
+          case TrusteeType.LEGAL_ENTITY:
+            hasTrusteeForChange = (appData[TrustKey]?.find(t => t.trust_id === trustId)?.CORPORATES ?? []).some(t => t.id === trusteeId);
+            break;
```

Comment:
would be good to have a default case that raises an error. 
default:  throw createAndLogErrorRequest(req, "Invalid trustee type");